src/Pact/Analyze/Eval/Numerical.hs

```diff
@@ -1,17 +1,26 @@
-{-# LANGUAGE FlexibleContexts      #-}
-{-# LANGUAGE Rank2Types            #-}
-{-# LANGUAGE ScopedTypeVariables   #-}
-{-# LANGUAGE TypeFamilies          #-}
-{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE Rank2Types          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE TypeOperators       #-}
 module Pact.Analyze.Eval.Numerical where
 
 import           Control.Lens            (over)
 import           Data.SBV                (EqSymbolic ((.==)), SymWord, sDiv,
-                                          sMod, (.^))
+                                          sMod, (.<), (.^))
 
 import           Pact.Analyze.Errors
-import           Pact.Analyze.Types.Eval
 import           Pact.Analyze.Types
+import           Pact.Analyze.Types.Eval
+
+-- This module handles evaluation of unary and binary integers and decimals.
```

Comment:
Property testing caught two problems in this module:

1. We weren't rounding all arithmetic ops.
2. We weren't marking failure on division by 0 or negative precision.

Reply:
🎉